src/ansi-c/CMakeLists.txt

```diff
@@ -5,9 +5,17 @@ add_executable(converter library/converter.cpp)
 
 file(GLOB ansi_c_library_sources "library/*.c")
 
-add_custom_command(OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/cprover_library.inc"
-    COMMAND cat ${ansi_c_library_sources} | $<TARGET_FILE:converter> > "${CMAKE_CURRENT_BINARY_DIR}/cprover_library.inc"
-    DEPENDS converter ${ansi_c_library_sources})
+# Convert cmake paths to native paths for Windows builds
+file(TO_NATIVE_PATH "${ansi_c_library_sources}"
+     ansi_c_library_sources_paths)
+file(TO_NATIVE_PATH "${CMAKE_CURRENT_BINARY_DIR}/cprover_library.inc"
+     cprover_library_path)
+file(TO_NATIVE_PATH "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/converter"
+     converter_path)
+
+add_custom_command(OUTPUT "${cprover_library_path}"
+    COMMAND cat ${ansi_c_library_sources_paths} | ${converter_path} > ${cprover_library_path}
```

Comment:
I'm afraid this doesn't quite work on windows (not on our CodeBuild machines here); We'd need `Get-Content`  here or add a `cat` alias for `Get-Content` in the codebuild.

Frustratingly this isn't showing up as a failure in codebuild due to an unrelated bug in how that build is set up. I've been thinking of changing the build so it publishes the build log somewhere publicly visible at least, but I've note gotten around to that yet.

Reply:
I'm not sure what to do here.  The add_custom_command is not my contribution.  My contribution is invocations of TO_NATIVE_PATH to get the right path separators on windows so the invocation of cat won't fail.  

Is the issue that we are using different instances of cat?  Maybe you are getting the cat that comes with cygwin that wants linux path separators and I am getting the cat that comes with the git shell that on windows wants windows path separators?

Personally, I think that if we get rid of cygwin completely and just build from the native command shell, that is the right way to go.

Reply:
On our codebuild jobs we're actually building on powershell, not a cygwin environment. The `TO_NATIVE_PATH` actually shouldn't make a difference for the most part; All windows builtins I'm aware of understand POSIX path separators.

Reply:
I think the natural thing to do is to build cbmc with the natural build environment on windows.  I have been trying to get cbmc to build within the visual studio command prompt that is just cmd with all the paths set correctly to run visual studio.  Visual studio comes with cmake and ninja.  Visual studio can be installed with git and the git shell that gives you access to all the linux utilities you might need.  I gave my procedure in the top comment of this pull request.  

We can simulate this inside an ordinary cmd shell, just run the script vcvars64.bat to set up the environment variables as the visual studio command shell does.

Let's stop doing all this crazy stuff to cbmc to build, and just get it to build the right way in the native build environment.

Reply:
This is exactly what we're already doing. https://github.com/diffblue/cbmc/blob/develop/buildspec-windows-cmake.yml

Reply:
> This is exactly what we're already doing. https://github.com/diffblue/cbmc/blob/develop/buildspec-windows-cmake.yml

Though, as far as I'm aware, this doesn't actually work? I believe this PR is a way towards making this work? Of course, we should discuss each individual change to see whether it's the right/best way to make things work.